### DISCUSSION TITLE
FormKit type file listening to the delete event

### DISCUSSION THREAD

**Discussion by fdverwoerd** [formkit/formkit#1027] [General]
If I have a:

```
<template>
  <FormKit v-for='f in extraFields'
    :type="f.type"
    label="f.name"
  :value="[{ name: 'pdf-example-1.pdf' }, { name: 'chocolate-recipe.docx' }]"
    help="if i remove all or 1 file, is there anyway to listen to the delete event?"
  />
</template>
```
And I click on remove all, or delete 1 file. The file gets deleted in the model. But I want to listen to this delete. Because I have a piece of code that uses @change and listen to file input and transform file into:

```
filesArray.value = [{
name: 'name.pdf',
content: 'base64'
}]
```

But After deleting the files I still have this array with the base64 copy. I am looking for a simple way to keep this in sync.

And another thing is that my FormKit type file is dynamic in a v-for loop. So this can either be text or file.

https://formkit.link/4f595dd01e586eacb7dbde4fd97e9019

**Comment by neobdev** [comment]
try this, hope it work

```
<FormKit
  type="file"
  :sections-schema="{
    fileRemove: {
      attrs: {
        onclick: (e) => {console.log(e);},
      `},`
    }
  }"/>
```

**Comment by sashamilenkovic** [comment]
I'm not sure this completely answers your question, but you could get the FormKit node from your file input and listen to the commit event: https://formkit.link/c55135a36356d86bfb9671f563781719 @fdverwoerd
